public/request/achievement/update-video.php

```diff
@@ -2,18 +2,52 @@
 
 use Illuminate\Support\Arr;
 use Illuminate\Support\Facades\Validator;
+use LegacyApp\Community\Enums\ArticleType;
+use LegacyApp\Platform\Enums\AchievementType;
+use LegacyApp\Platform\Models\Achievement;
 use LegacyApp\Site\Enums\Permissions;
 
-if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Developer)) {
+if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::JuniorDeveloper)) {
     abort(401);
 }
 
 $input = Validator::validate(Arr::wrap(request()->post()), [
-    'achievement' => 'required|integer',
+    'achievement' => 'required|integer|exists:mysql_legacy.Achievements,ID',
     'video' => 'nullable|url',
 ]);
 
-if (updateAchievementEmbedVideo((int) $input['achievement'], $input['video'])) {
+$achievementId = (int) $input['achievement'];
+$embedUrl = $input['video'];
+
+$achievement = Achievement::find($achievementId);
+if (!$achievement) {
+    abort(400);
+}
```

Comment:
Missing record should be 404. Also, doesn't the `exists` validation check ensure this will never happen?

Reply:
Good point, resolved in https://github.com/RetroAchievements/RAWeb/pull/1526/commits/7f73a9c84c80d856ead8fa30952ada086a51f44e.